sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnSignaturesTest.java

```diff
@@ -132,14 +135,100 @@ void finishBundle() {}
   }
 
   @Test
+  public void testTimerIdWithWrongType() throws Exception {
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("TimerId");
+    thrown.expectMessage("TimerSpec");
```

Comment:
`thrown.expectMessage("bizzle");` so that we verify we also point at the problem?